scripts/cluster/agent.py

```diff
@@ -423,6 +443,285 @@ def configure():
+    return output
+
+
+@app.route('/{}/addon/disable'.format(CLUSTER_API), methods=['POST'])
+def disable():
+    """
+    Web call to microk8s.disable <addon>
+    :request_param addon: the name of the addon
+    """
+    v = callback_token_validation(request)
+    if not v["valid"]: return v["response"]
+    if "addon" not in request.json or len(request.json["addon"].strip()) == 0:
+        error_msg = {"error": "Empty addon provided"}
+        return Response(json.dumps(error_msg), mimetype='application/json', status=400)
+    output = subprocess.check_output("{}/microk8s-disable.wrapper {}".format(snap_path, request.json["addon"]).split())
+    return output
+
+
+@app.route('/{}/start'.format(CLUSTER_API), methods=['POST'])
+def start():
+    """
+    Web call for microk8s.start
+    """
+    v = callback_token_validation(request)
+    if not v["valid"]: return v["response"]
+    output = subprocess.check_output("{}/microk8s-start.wrapper".format(snap_path).split())
+    return output
+
+
+@app.route('/{}/stop'.format(CLUSTER_API), methods=['POST'])
+def stop():
+    """
+    Web call for microk8s.stop
+    """
+    v = callback_token_validation(request)
+    if not v["valid"]: return v["response"]
+    output = subprocess.check_output("{}/microk8s-stop.wrapper".format(snap_path).split())
+    return output
+
+
+@app.route('/{}/overview'.format(CLUSTER_API), methods=['POST'])
+def overview():
+    """
+    Web call to get the microk8s.kubectl get all --all-namespaces
+    """
+    v = callback_token_validation(request)
+    if not v["valid"]: return v["response"]
+    output = subprocess.check_output("{}/microk8s-kubectl.wrapper get all --all-namespaces".format(snap_path).split())
+    return output
+
+
+@app.route('/{}/status'.format(CLUSTER_API), methods=['POST', 'GET'])
+def status():
+    """
+    Web call to get the microk8s status
+    """
+    cmd = "{}/microk8s-status.wrapper --format yaml --timeout 60".format(snap_path)
+    if request.method.upper() == 'GET' and not is_extended_api_enabled():
+        error_msg = {"error": ext_api_error_message.format('GET')}
+        return Response(json.dumps(error_msg), mimetype='application/json', status=400)
+    if request.method.upper() == 'POST':
+        v = callback_token_validation(request)
+        if not v["valid"]: return v["response"]
+        if "addon" in request.json:
+            cmd = "{}/microk8s-status.wrapper -a {}".format(snap_path, request.json["addon"])
+
+    output = subprocess.check_output(cmd.split())
+
+    if request.json and "addon" in request.json:
+        json_output = {"addon": request.json["addon"], "status": output.decode().strip('\n')}
+    else:
+        json_output = yaml.full_load(output)
+
+    return app.response_class(response=json.dumps(json_output, sort_keys=False, indent=4), status=200, mimetype='application/json')
+
+
+@app.route('/{}/config'.format(CLUSTER_API), methods=['POST'])
```

Comment:
I do not think you need this for MicroDash. Please make a case why this is needed in a separate PR.

Reply:
I may have a use case if this method returns the kubeconfig? 
If yes, it can be helpful when working with clusterapi when u have to return the kubeconfig through kubernetes secret.

Reply:
This is only accessible if you 'enable-extended-api', passing also the callback-token using a Header or the json body